plugins/mcreator-core/blockly/js/mcreator_extensions.js

```diff
@@ -37,6 +37,47 @@ Blockly.Extensions.register('is_custom_loop',
         Blockly.libraryBlocks.loops.loopTypes.add(this.type);
     });
 
+Blockly.Extensions.register('procedure_dependencies_tooltip',
+    function () {
+        let thisBlock = this;
+        this.setTooltip(function () {
+            const depList = javabridge.getAllDependencies(thisBlock.getFieldValue('procedure'));
+            if (depList.length == 0)
+                return javabridge.t('blockly.extension.procedure_dep_tooltip.empty');
+            let tooltip = javabridge.t('blockly.extension.procedure_dep_tooltip');
+            for (const dependency of depList)
+                tooltip += '\n' + dependency;
+            return tooltip;
+        });
+    });
+
+Blockly.Extensions.register('procedure_dependencies_onchange_mixin',
+    function () {
+        this.setOnChange(function (changeEvent) {
+            // Trigger the change only if a block is changed or created
+            if (changeEvent.type !== Blockly.Events.BLOCK_CHANGE &&
+                changeEvent.type !== Blockly.Events.BLOCK_CREATE) {
+                return;
+            }
+            const group = Blockly.Events.getGroup();
+            // Makes it so the block change and the unplug event get undone together.
+            Blockly.Events.setGroup(changeEvent.group);
+            // For each passed input, we check if it's within bounds
+            for (var i = 0; this.getField('name' + i); i++) {
+                const prevType = this.getInput('arg' + i).connection.getCheck();
+                this.getInput('arg' + i).setCheck(javabridge.getDependencyType(
+                    this.getFieldValue('procedure'), this.getFieldValue('name' + i)));
+                const newType = this.getInput('arg' + i).connection.getCheck();
+                if (changeEvent.type !== Blockly.Events.BLOCK_CREATE && ((prevType && newType) ?

```

Comment:
This condition could be simplified:
- The event is either CREATE or CHANGE, so this could check directly if it's `BLOCK_CHANGE`
- `null` can be passed to `JSON.stringify`, so there's no need to handle it separately

Having some more comments would also be nice

Reply:
Doesn't `BlockCreate` extend `BlockBase` event, not `BlockChange`?

Reply:
Fine, I'm blind 👀

Reply:
✔️ Fixed